File: Js/Binaryalgo.js
const Binaryfunc = (arr, target) => {
  // Initiate start and end
  let start = 0
  let end = arr.length - 1
  // till end is less then start
  while (start <= end) {
    // Mid is start , end-start is that
    // that makes the things
    let mid = start + (end - start) / 2

    // target is less then mid,move end to less then mid
    if (target < arr[mid]) {
      end = mid - 1
      // target is more then mid,move start with mid plus one
    } else if (target > arr[mid]) {
      start = mid + 1
    } else {
      // Less then mid,
      return mid
    }
  }
  //   If not found return -1
  return -1
}
arr = [1, 2, 6, 10, 15, 19, 21]
target = 15
Binaryfunc(arr, 15)

File: python/Binaryalgo.py
from tokenize import Number


class BinaryAlgo:
    def __init__(self, arr, target) -> Number:
        self.arr = arr
        self.target = target

    def binarySearch(self):
        start = 0
        end = len(self.arr)-1

        while(start <= end):
            mid = start+(end-start)//2

            if(self.target < self.arr[mid]):
                end = mid-1
            elif(self.target > self.arr[mid]):
                start = mid+1
            else:
                print(mid)
                return mid
        return -1


arr = [1, 3, 5, 6, 7, 9, 15, 18]
target = 15
le = BinaryAlgo(arr, target)
le.binarySearch()
